core/matrix/dense_kernels.hpp

```diff
@@ -34,10 +34,12 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define GKO_CORE_MATRIX_DENSE_KERNELS_HPP_
 
 
-#include <ginkgo/core/base/types.hpp>
 #include <ginkgo/core/matrix/dense.hpp>
 
 
+#include <ginkgo/core/base/types.hpp>
+
```

Comment:
The standard headers such as `<memory>` and possibly others are missing here. Is it possible to fix the proper standard headers here and other files as well ? Or would you prefer it be done in another PR ?

Reply:
I think it is better in another PR (maybe including `include-what-you-use`).
For the `include-what-you-use`, add the recommdation in the begin and remove the unneeded header. Moreover, always gives warning to check it again.
If there is a #if-#else block, it is hard to insert them at the correct position.

Reply:
`include-what-you-use` is sometimes wrong, so I would not directly include its result as part of a script. Maybe it can be configured to always provide the correct solution, but this requires some work.

See this for example https://gitlab.com/ginkgo-project/ginkgo-public-ci/-/jobs/404997031#L2851
`name_demangling` is required, we use in stream `name_demangling::get_dynamic_type`, but it says it should be removed...